modules/rostests/apitests/shlwapi/PathUnExpandEnvStringsForUser.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * PROJECT:         ReactOS api tests
+ * LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
+ * PURPOSE:         Tests for PathUnExpandEnvStringsForUser
+ * PROGRAMMERS:     Katayama Hirofumi MZ
+ */
+
+#include <apitest.h>
+#include <shlwapi.h>
+#include <stdio.h>
+
+#define DO_TEST(Res, hToken, TestStr, ExpStr, Len) \
+do { \
+    BOOL ret = PathUnExpandEnvStringsForUserW((hToken), (TestStr), OutStr, Len); \
+    ok(ret == (Res), "Tested %s, expected returned value %d, got %d\n", \
+       wine_dbgstr_w((TestStr)), (Res), ret); \
+    if (ret) \
+        ok(_wcsicmp(OutStr, (ExpStr)) == 0, "Tested %s, expected %s, got %s\n", \
+           wine_dbgstr_w((TestStr)), wine_dbgstr_w((ExpStr)), wine_dbgstr_w(OutStr)); \
+} while (0)
+
+// PathUnExpandEnvStringsForUserW
+typedef BOOL (WINAPI *PATHUNEXPANDENVSTRINGSFORUSERW)(HANDLE hToken, LPCWSTR pszPath, LPWSTR pszUnExpanded, INT cchUnExpanded);
+PATHUNEXPANDENVSTRINGSFORUSERW pPathUnExpandEnvStringsForUserW = NULL;
+
+START_TEST(PathUnExpandEnvStringsForUser)
+{
+    DWORD ret;
+    WCHAR OutStr[MAX_PATH], TestStr[MAX_PATH];
+    HINSTANCE hShlwapi;
+
+    hShlwapi = LoadLibraryA("shlwapi");
```

Comment:
You could just use GetModuleHandleW here and save yourself the cleanup (and maybe even error check). This test statically imports shlwapi.dll, so it's guaranteed to be loaded already.

Reply:
OK.